test/test_trees_and_forests.jl

```diff
@@ -0,0 +1,38 @@
+using Test
+using Graphs
+using NamedGraphs
+using NamedGraphs: forest_cover, spanning_tree
+
+module TestTreesAndForests
+using NamedGraphs
+using NamedGraphs: hexagonal_lattice_graph, triangular_lattice_graph
+gs = [
+  ("Chain", named_grid((6, 1))),
+  ("Cubic Lattice", named_grid((3, 3, 3))),
+  ("Hexagonal Grid", hexagonal_lattice_graph(6, 6)),
+  ("Comb Tree", named_comb_tree((4, 4))),
+  ("Square lattice", named_grid((10, 10))),
+  ("Triangular Grid", triangular_lattice_graph(5, 5; periodic=true)),
+]
+algs = [NamedGraphs.BFS(), NamedGraphs.DFS(), NamedGraphs.RandomBFS()]
+end
+
+@testset "Test Spanning Trees $g_string, $alg" for (g_string, g) in TestTreesAndForests.gs,
+  alg in TestTreesAndForests.algs
+
+  s_tree = spanning_tree(alg, g)
```

Comment:
```suggestion
  s_tree = spanning_tree(g; alg)
```
Let's consider the kwarg syntax to be the external API, so the tests should test that interface.